fe/fe-core/src/main/java/com/starrocks/http/BaseAction.java

```diff
@@ -353,6 +361,23 @@ private boolean parseAuthInfo(BaseRequest request, ActionAuthorizationInfo authI
         return true;
     }
 
+    // Refer to {@link #parseAuthInfo(BaseRequest, ActionAuthorizationInfo)}
+    public static ActionAuthorizationInfo parseAuthInfo(String fullUserName, String password, String host) {
+        ActionAuthorizationInfo authInfo = new ActionAuthorizationInfo();
+        final String[] elements = fullUserName.split("@");
```

Comment:
I suggest you use `com.starrocks.analysis.UserIdentity#fromString`.

Reply:
Here we want to split `fullUserName` into a user name and a cluster id, but `UserIdentity#fromString` want to get a user name and a host. I think they are different, and what do you think?

Reply:
Got it.